vcd/resource_vcd_nsxt_edgegateway_l2_vpn_tunnel.go

```diff
@@ -0,0 +1,399 @@
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceVcdNsxtEdgegatewayL2VpnTunnelImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"edge_gateway_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Edge gateway ID for the tunnel",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of the L2 VPN Tunnel session",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Description of the L2 VPN Tunnel session",
+			},
+			"session_mode": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"CLIENT", "SERVER"}, false),
+				ForceNew:     true,
+				Description:  "Mode of the tunnel session, must be CLIENT or SERVER",
+			},
+			"enabled": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "Status of the L2 VPN Tunnel session. Always set to `true` for CLIENT sessions",
+			},
+			"local_endpoint_ip": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "Local endpoint IP of the tunnel session, the IP must be sub-allocated to the Edge Gateway",
+				ValidateFunc: validation.IsIPAddress,
+			},
+			"remote_endpoint_ip": {
+				Type:     schema.TypeString,
+				Required: true,
+				Description: "The IP address of the remote endpoint, which corresponds to the device" +
+					"on the remote site terminating the VPN tunnel.",
+				ValidateFunc: validation.IsIPAddress,
+			},
+			"tunnel_interface": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				Description: "Network CIDR block over which the session interfaces. Only relevant if " +
+					"`session_mode` is set to `SERVER`",
+				ValidateFunc: validation.IsCIDR,
+			},
+			"connector_initiation_mode": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Description: "Connector initation mode of the session describing how a connection is made. " +
+					"Needs to be set only if `session_mode` is set to `SERVER`",
+				ValidateFunc: validation.StringInSlice([]string{"INITIATOR", "RESPOND_ONLY", "ON_DEMAND"}, false),
+			},
+			"pre_shared_key": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Description: "Pre-shared key used for authentication, needs to be provided only on creation of" +
+					"`SERVER` sessions",
+			},
+			"peer_code": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				Description: "Base64 encoded string of the full configuration of the tunnel, " +
```

Comment:
In addition to the remarks given for the data source, this description leaves a doubt.
If it is a required input, does it mean that we have to encode it before passing it on?
If it is retrieved from another `vcd_nsxt_edgegateway_l2_vpn_tunnel`, does it come already encoded?

Reply:
I updated while trying to also specify when it needs to be provided. I can't think of a good way to describe the `base64` part it in a concise way, what I thought is to describe what the field contains by itself, but other than that, people would find actual usages in the examples of the documentation.